database/migrations/000018_claims_to_match.up.sql

```diff
@@ -0,0 +1,29 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+CREATE TABLE IF NOT EXISTS mapped_role_grants(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    project_id UUID NOT NULL REFERENCES projects(id) ON DELETE CASCADE,
+    role TEXT NOT NULL,
+    claim_mappings JSONB NOT NULL,
+    resolved_subject TEXT REFERENCES users(identity_subject) ON DELETE SET NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
```

Comment:
I really don't want to take the approach of tying fully to keycloak. We've already done something like that in several cases... and I'd really like to move away from that. If we do require more functionality from keycloak, I rather see us invest in tying up to it's eventing system and have it be pluggable, instead of making direct calls that get compiled in our source code.